app/controllers/add-preprint.js

```diff
@@ -1,87 +1,207 @@
 import Ember from 'ember';
+import loadAll from 'ember-osf/utils/load-relationship';
+import NodeActionsMixin from 'ember-osf/mixins/node-actions';
 
-export default Ember.Controller.extend({
+// Enum of available states
+export const State = Object.freeze(new Ember.Object({
+    START: 'start',
+    NEW: 'new',
+    EXISTING: 'existing'
+}));
+
+export default Ember.Controller.extend(NodeActionsMixin, {
     toast: Ember.inject.service(),
-    _url: null,
-    openModal: false,
+    session: Ember.inject.service(),
+    panelActions: Ember.inject.service(),
+    currentUser: Ember.inject.service(),
+    fileManager: Ember.inject.service(),
+
+    _State: State,
+    state: State.START,
+    uploadState: State.START,
+    uploadFile: null,
     resolve: null,
-    latestFileName: null,
-    preprint: null,
+    preprintNode: null,
+    user: null,
+    userNodes: Ember.A([]),
     dropzoneOptions: {
         uploadMultiple: false,
         method: 'PUT'
     },
+
+    _names: ['upload', 'basics', 'subjects', 'authors', 'submit'].map(str => str.capitalize()),
+    valid: new Ember.Object(),
+    init() {
+        this._super(...arguments);
+        if (this.get('session.isAuthenticated')) {
+            this._setCurrentUser();
+        }
+    },
+    _setCurrentUser() {
+        this.get('currentUser').load().then(user => this.set('user', user));
+    },
+    _refreshNodes() {
+        const user = this.get('user');
+        if (user) {
+            loadAll(user, 'nodes', this.get('userNodes'));
+        } else {
+            this.set('userNodes', Ember.A());
+        }
+    },
+    onGetCurrentUser: Ember.observer('user', function() {
+        this._refreshNodes();
+    }),
+
+    filter: [{}, {}, {}],
```

Comment:
The objects get a `value` key set on them with the filter keyword. This is to allow ember to observe changes in the array indices themselves with `filter.@each.value`.